mutters-core/src/main/java/com/rabidgremlin/mutters/core/MatcherScores.java

```diff
@@ -0,0 +1,69 @@
+/* Licensed under Apache-2.0 */
+package com.rabidgremlin.mutters.core;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Holds the matching scores generated by an IntentMatcher.
+ * 
+ * @author rabidgremlin
+ *
+ */
+public class MatcherScores
+{
+  /** Holds the matching scores. */
+  private SortedMap<Double, Set<String>> scores = new TreeMap<Double, Set<String>>();
+
+  /**
+   * Creates a MatcherScores containing no scores.
+   * 
+   */
+  public MatcherScores()
+  {
+    // do nothing
+  }
+
+  /**
+   * Create a MatcherScores with the supplied scores. Map is keyed by score to
+   * allow multiple intents to have the same score.
+   * 
+   * @param scores Map of scores mapped to sets of intent names.
+   */
+  public MatcherScores(SortedMap<Double, Set<String>> scores)
+  {
+    this.scores = scores;
+  }
+
+  /**
+   * Add a score for an intent to the matcher scores.
+   * 
+   * @param intent The name of the intent to add the score for.
+   * @param score  The score for the intent.
+   */
+  public void addScore(String intent, Double score)
+  {
+    Set<String> intents = scores.get(score);
+    if (intents == null)
+    {
+      intents = new HashSet<String>();
```

Comment:
can this be a ```LinkedHashSet```
otherwise hard to test because the order is unknown

Reply:
or ```TreeSet```
would also be good if it returned itself so can be used like a builder:

```
  public MatcherScores addScore(String intent, Double score) {
    scores.computeIfAbsent(score, k -> new TreeSet<>()).add(matchingIntent);
    return this;
}
```

Reply:
Implemented.